### ISSUE TITLE
Enumeration over Task Graph Results

### ISSUE THREAD

**Issue by radiantone** [dask/dask#5483]
There does not seem to be a way to cause a task graph specification to enumerate a results over a receiving function in parallel.

graph = {
    "documents": (thedocs,),
    "analylze": (analyzedoc, "documents")
}

I do not want to send the "list of documents" to analyzedoc. I want to essentially map each document in "documents" to a separate (parallel) call to analyedoc.

**Comment by mrocklin** [comment]
See https://examples.dask.org/applications/evolving-workflows.html

**Comment by mrocklin** [comment]
Short term that's probably out of scope for Dask.  When people ask for ways
to compute evolving graphs we recommend that they use one of the three
approaches in that example.

**Comment by radiantone** [comment]
Gotcha. The net result however, is that we can only get parallelism on task graphs that return non-list results. If a collection or list is returned, then a function has to process that list itself - reducing the point of task graphs. Mainly I see this as an arity concern in the computational model. And to be complete, i should support one to many mapping across a function.

**Comment by mrocklin** [comment]
You're welcome to give it a try.  I think that you'll find it challenging to get what you want and also satisfy all other usage constraints though.

**Comment by radiantone** [comment]
I actually did it before for a project. I wrote a JSON dialect for determining the graph flow, arguments and metadata such as parallelism and flow control indicators. Then a process compiles that into a dask task using function wrappers to dispatch calls to functions, gather results, etc. But it does require some scaffolding that runs outside the graph. Something that would be better if it wasn't necessary.
